### ISSUE TITLE
Problems playing video

### ISSUE THREAD

**Issue by xorinzor** [savonet/liquidsoap#1245]
I'm getting this output https://pastebin.com/n40X5D06 with a container built using https://github.com/xorinzor/shoutz0r-docker/blob/master/Dockerfile when it tries to play the `fallback.mkv` file.

When I changed the file to `fallback_new.mp4` I got this output https://pastebin.com/tzNGrF7b

Tried installing gstreamer 1.0 separately as well, but that resulted in liquidsoap not detecting gstreamer.

Why is this happening, and is there anything I'm missing?

[video files.zip](https://github.com/savonet/liquidsoap/files/4774775/video.files.zip)
[wrapper.liq.txt](https://github.com/savonet/liquidsoap/files/4774777/wrapper.liq.txt)

**Comment by toots** [comment]
Hi! How did you install gstreamer? Do you have all gstreamer plugins installed? It looks like the demuxer for `mp4` is installed with `gst-plugins-good` etc.

**Comment by toots** [comment]
The other approach to decouple from liquidsoap is to try to run a manual pipeline using `gst-launch` and see if you can get gstreamer to decode a file. If that's the case, then liquidsoap decoder should work unless it is indeed a bug with us.

**Comment by xorinzor** [comment]
@toots can you explain how I can do so using `opam`? 

The OPAM repository only shows the main gstreamer package (albeit very outdated at 0.3 instead of 1.0) and none of the plugin packages.

As shown in the dockerfile I had installed pretty much all liquidsoap-dependencies as written at https://www.liquidsoap.info/doc-1.4.2/build.html in the hopes of avoiding these problems :/

The dockerfile shows exactly what I'm currently using, I figured it was easier to show that way, then to try and explain.

**Comment by toots** [comment]
Gstreamer plugins install depend on the system and are not installed via `opam`. What system is your docker file using?

**Comment by xorinzor** [comment]
I'm using docker for windows (with linux containers, not windows containers) and I manage them using Dockstation.

The dockerfile is using Debian 10, who's sources.list I have updated to use the (debian 10) "buster" repository (for some reason it kept using (debian 9) "stretch" repositories for me).

I can manually install gstreamer, but since liquidsoap doesn't detect gstreamer once I do, I'll need a way to tell liquidsoap where to find gstreamer

**Comment by toots** [comment]
You shouldn't need to manually install gstreamer. The package for the `good` plugins on debian is here, typically: https://packages.debian.org/jessie/gstreamer1.0-plugins-good

You might need to also install more plugins.

**Comment by xorinzor** [comment]
So, I overhauled my Dockerfile to use Sudo and Opam the correct way, which already improved things a lot.

I also simplified my `wrapper.liq` file to make the testing easier.

```
set("audio.converter.samplerate.libsamplerate.quality", "best")

set("frame.video.width",1920)
set("frame.video.height",1080)
set("frame.video.samplerate",60)

set("decoder.debug",true)
set("log.level",5)

#Create our unfallible source
fallback_source = single("/home/liquidsoap/app/resources/videoplayback.mp4")

videoEncodingFormat = %gstreamer(channels=2,
                 audio="lamemp3enc",
                 has_video=true,
                 video="x264enc speed-preset=6",
                 muxer="mpegtsmux",
                 metadata="metadata",
                 log=5)

#stream output
output.icecast(
  videoEncodingFormat,
  mount = "/test",
  id = "shoutzor",
  host = "172.22.0.2",
  port = 8000,
  format="video/x-matroska",
  password = "changeme",
  icy_metadata = "false",
  fallback_source)
```

There's still something going wrong however. It keeps complaining about the wrong number of channels and it being an invalid mime-type. Yet all modules correctly detect the type of video and audio stream. Somewhat later on it suddenly also talks about it being an invalid URI?

![Capture](https://user-images.githubusercontent.com/4212335/85203523-b1ac6200-b30e-11ea-946b-caa72b4120c2.PNG)
![Capture 2](https://user-images.githubusercontent.com/4212335/85203526-b3762580-b30e-11ea-87e3-ba38e95caaf9.PNG)

**Comment by xorinzor** [comment]
boop @toots

**Comment by toots** [comment]
Appart from the current `master`, no decoders in released liquidsoap versions (last one 1.4.2 as of writing this) can decode `mp4` except `gstreamer` so the first logs are to be expected.

Now, regarding `gstreamer`, I suggest to try first decoding outside of liquidsoap. Something like this:
```
gst-play-1.0 /path/to/file
```
and see if it correctly decodes the file. If no, then something is up with your gstreamer installation.

As a side note, I would recommend experimenting with the `master` branch. I wouldn't say that it is fully ready for production work as of now but we're actively working on it and the code has much more solid support for decoding and encoding of video media using `ffmpeg`.

**Comment by xorinzor** [comment]
@toots took me a while to get it to work by pinning the repositories, but eventually it finally compiled :)

Running the script on 1.5.0 I see a lot of improvement, but still some vague errors, and the detection doesn't seem to work properly either. Since you said mp4 wasn't supported in 1.4.2, I figured I'd also try with an avi file, just for comparison. Both video types had the same result.

I should add that the script kept running and had no clock errors while it wasn't able to connect to the icecast mount. Once I fixed that and it was able to connect, this clock error started to appear.

These are screenshots of the incorrect detection and clock error:

![Capture](https://user-images.githubusercontent.com/4212335/86174334-c0222700-bb21-11ea-8486-a52499edf955.PNG)
![Capture 2](https://user-images.githubusercontent.com/4212335/86174345-c2848100-bb21-11ea-9030-0ce418eebac2.PNG)
